File: gin/library/tencent/trtc.go
package tencent

import (
	"webmis/config"
	"webmis/library"
	"webmis/util"
)

/* 实时音视频 */
type Trtc struct {
	Signature
	SdkAppId int
}

/* 公共配置 */
func (t *Trtc) Init() {
	t.New()
	cfg := config.TRTC()
	t.ApiUrl = "https://trtc.tencentcloudapi.com/"
	t.Host = "trtc.tencentcloudapi.com"
	t.Service = "trtc"
	t.Version = "2019-07-22"
	t.SdkAppId = cfg.SDKAppID
}

/* 房间-查询 */
func (t *Trtc) RoomList(StartTime int64, EndTime int64) map[string]interface{} {
	// 参数
	t.Init()
	t.Action = "DescribeRoomInformation"
	// 数据
	data := map[string]interface{}{
		"SdkAppId":  (&util.Type{}).Strval(t.SdkAppId),
		"StartTime": StartTime,
		"EndTime":   EndTime,
	}
	// 请求头
	header := t.V3Header(data)
	byt, _ := (&library.Curl{}).Request(t.ApiUrl, util.JsonEncode(data), "POST", header)
	res := map[string]interface{}{}
	util.JsonDecode(byt, &res)
	return res
}

/* 房间-解散 */
func (t *Trtc) RoomDismiss(roomId string) map[string]interface{} {
	// 参数
	t.Init()
	t.Action = "DescribeRoomInformation"
	// 数据
	data := map[string]interface{}{
		"SdkAppId": (&util.Type{}).Strval(t.SdkAppId),
		"RoomId":   roomId,
	}
	// 请求头
	header := t.V3Header(data)
	byt, _ := (&library.Curl{}).Request(t.ApiUrl, util.JsonEncode(data), "POST", header)
	res := map[string]interface{}{}
	util.JsonDecode(byt, &res)
	return res
}

File: java/src/main/java/webmis/library/tencent/Trtc.java
package webmis.library.tencent;

import java.util.HashMap;

import com.alibaba.fastjson.JSONObject;

import webmis.config.Tencent;
import webmis.library.Curl;

/* 实时音视频 */
public class Trtc extends Signature {

  public static int SdkAppId = 0;

  /* 公共配置 */
  public static void Init() {
    HashMap<String, Object> cfg = Tencent.TRTC();
    ApiUrl = "https://trtc.tencentcloudapi.com/";
    Host = "trtc.tencentcloudapi.com";
    Service = "trtc";
    Version = "2019-07-22";
    SdkAppId = Integer.valueOf(cfg.get("SDKAppID").toString());
  }

  /* 房间-查询 */
  public static JSONObject RoomList(long StartTime, long EndTime) {
    // 参数
    Init();
    Action = "DescribeRoomInformation";
    // 数据
    JSONObject data = new JSONObject();
    data.put("SdkAppId", String.valueOf(SdkAppId));
    data.put("StartTime", StartTime);
    data.put("EndTime", EndTime);
    // 请求头
    HashMap<String, Object> header = V3Header(data);
    return Curl.Request(ApiUrl, data.toString(), "POST", header);
  }

  /* 房间-解散 */
  public static JSONObject RoomDismiss(String roomId) {
    // 参数
    Init();
    Action = "DismissRoomByStrRoomId";
    // 数据
    JSONObject data = new JSONObject();
    data.put("SdkAppId", String.valueOf(SdkAppId));
    data.put("RoomId", roomId);
    // 请求头
    HashMap<String, Object> header = V3Header(data);
    return Curl.Request(ApiUrl, data.toString(), "POST", header);
  }
  
}
